scala/lambdas/ingest-failure-notifications/src/main/scala/uk/gov/nationalarchives/ingestfailurenotifications/Lambda.scala

```diff
@@ -0,0 +1,64 @@
+package uk.gov.nationalarchives.ingestfailurenotifications
+
+import cats.effect.IO
+import io.circe.DecodingFailure.Reason.CustomReason
+import io.circe.generic.semiauto.deriveDecoder
+import io.circe.{Decoder, DecodingFailure, HCursor}
+import io.circe.parser.decode
+import pureconfig.ConfigReader
+import pureconfig.generic.derivation.default.*
+import uk.gov.nationalarchives.{DADynamoDBClient, DASNSClient}
+import uk.gov.nationalarchives.utils.LambdaRunner
+import uk.gov.nationalarchives.ingestfailurenotifications.Lambda.{*, given}
+import uk.gov.nationalarchives.DADynamoDBClient.given
+import org.scanamo.syntax.*
+import uk.gov.nationalarchives.dynamoformatters.DynamoFormatters.IngestLockTableItem
+import uk.gov.nationalarchives.utils.ExternalUtils.MessageStatus.IngestError
+import uk.gov.nationalarchives.utils.EventCodecs.given
+import uk.gov.nationalarchives.utils.ExternalUtils.MessageType.IngestUpdate
+import uk.gov.nationalarchives.utils.ExternalUtils.{OutputMessage, OutputParameters, OutputProperties}
+
+import java.time.Instant
+import java.util.UUID
+
+class Lambda extends LambdaRunner[SfnEvent, Unit, Config, Dependencies]:
+  override def handler: (SfnEvent, Config, Dependencies) => IO[Unit] = (event, config, dependencies) =>
+    for {
+      items <- dependencies.dynamoClient
+        .queryItems[IngestLockTableItem](config.lockTableName, "groupId" === event.detail.input.groupId, Option(config.lockTableGsiName))
+      messages = items.map { item =>
+        OutputMessage(
+          OutputProperties(event.detail.input.batchId, dependencies.uuidGenerator(), None, Instant.now, IngestUpdate),
```

Comment:
So does this mean that all items we get from Dynamo (via the `groupId`) will have the same `batchId`?

Reply:
What we're sending here is the batchId in the TDR_<uuid>_0 format. Because this is triggered by the step function failure, we know that all the messages sent will be for the batch that step function is processing if that makes sense. 
The batchId is only linked to the lock table in the sense that that's the ID of the currently running step function.

Reply:
Ah, so even though we are getting the items from dynamo based on the `groupId`, they will all have the same `batchId` because the locktable on holds what's actually running and therefore you can't items belonging to both `_0` and `_1` batches at the same time?

Reply:
Yeah, exactly.